### PR TITLE
feat: [#765] Add pid handling for: HANDLE, PMID and PMC

### LINKED ISSUE #765: Fix the PIDs in the research products view
Currently, we use the DOI as a filter on the research products box view (this is a bad attitude)

Acceptance criteria:
- [x] As a User I would like to have the DOI as a link to an external page (each DOI should have the external link)
- [x] As a User I would like to see other identificatory if exists. (these identificators should look and behave as DOI)

List of identificators:
- HANDLE
- .DOI
- PMID
- PMD

Figma: https://www.figma.com/file/pTiscgrb4ie8R57vJmEhyL/Boxy-Redesign?type=design&node-id=7-2879&mode=design&t=3kJavuus7ieNTgsJ-4

### PR DESCRIPTION
**Pull request by wujuu**
Closes #765.

Also added a hotfix for doctype being visible again in publications.

### CODE REVIEW

**Review by Anok666 — APPROVED** [review]

File: ui/apps/ui/src/app/collections/data/utils.ts [hunk 1]
```diff
@@ -70,21 +71,65 @@ export const formatPublicationDate = (
...
   const doiRegex = new RegExp('^10\\.\\d{4,9}/[-._;()/:A-Za-z0-9]+$');
   return doiRegex.test(rawDoi) ? `https://doi.org/${rawDoi}` : undefined;
 };
 
-export const constructDoiTag = (
-  doiArr: string[] | undefined
+const resolveHandleLink = (rawHandle: string): string | undefined => {
+  const handleRegex = new RegExp('^\\S+/\\S+$');
+  return handleRegex.test(rawHandle)
+    ? `https://hdl.handle.net/${rawHandle}`
```
Comment (Michal-Kolomanski) [inline ui/apps/ui/src/app/collections/data/utils.ts:82]: Current PID:
- DOI: `https://doi.org/${rawDoi}`

This PR added:
- HANDLE: `https://hdl.handle.net/${rawHandle}`
- PMC: `https://europepmc.org/article/PMC/${urlPart}`
- PMID: `https://pubmed.ncbi.nlm.nih.gov/${rawPmid}`

Total statistics for PIDs:
```
has_doi     3947182
no pid      1093032
has_handle   169408
has_pmid      53261
has_pmc       23839
has_arXiv     12011
has_w3id       3051
has_pdb           2
```
- Shouldnt we also add arXiv and w3id PID? @agpul @wujuu 

@Marcelinna @agpul 
The PIDs that have been added, need to be tested on this PR. Furthermore, it is necessary to confirm that those links are proper for each PID. Maybe Anna Wałek could confirm this?

Comment (wujuu) [reply ui/apps/ui/src/app/collections/data/utils.ts:82]: Based on @wiktorflorian 's research - those as the correct linka for these pids. We basically haven't found any counter examples where they wouldn't work

**Review by Michal-Kolomanski — APPROVED** [review]
Good job! @Marcelinna please check if those new added PIDs are visible and working

**Comment by Marcelinna** [comment]
I think we don't want to merge this if the HANDLE identifier is so extensive @agpul @aleksandrahojszyk @wujuu 
image

**Comment by Marcelinna** [comment]
@agpul @wujuu I don't see PMC in ticket
image

**Comment by agpul** [comment]
> @wujuu Can you provide me a record with a PMD identifier? image

Hmm I think it is a misspelling PMC, not PMD, sorry.

**Comment by agpul** [comment]
> @wujuu Can you provide me a record with a PMD identifier? image

same here PMC not PMD

**Comment by agpul** [comment]
> I think we don't want to merge this if the HANDLE identifier is so extensive @agpul @aleksandrahojszyk @wujuu image https://search-4.docker-fid.grid.cyf-kr.edu.pl/search/dataset?q=*&standard=false

@aleksandrahojszyk this need to be fixed somehow

**Comment by agpul** [comment]
> > I think we don't want to merge this if the HANDLE identifier is so extensive @agpul @aleksandrahojszyk @wujuu image https://search-4.docker-fid.grid.cyf-kr.edu.pl/search/dataset?q=*&standard=false
> 
> @aleksandrahojszyk this need to be fixed somehow

we agreed to add a show more button if there are more than 4 identifiers.

**Comment by aleksandrahojszyk** [comment]
Screenshot 2023-09-15 at 10 19 15

**Comment by wujuu** [comment]
@Marcelinna @aleksandrahojszyk I've added "show more" feature to all tags - but it's only enabled in identifiers for now.

Also changed the regex for HANDLE because some of the datasets had straight-up url instead of an identifier. We can click on that as well now.

**Comment by Marcelinna** [comment]
> @Marcelinna @aleksandrahojszyk I've added "show more" feature to all tags - but it's only enabled in identifiers for now.
> 
> Also changed the regex and url construction for HANDLE because some of the datasets had straight-up url instead of an identifier. We can click on that as well now.

@aleksandrahojszyk we want add this to all tags ?
image

**Comment by Marcelinna** [comment]
@wujuu  the same color?
image

**Comment by aleksandrahojszyk** [comment]
SORRY I made a mistake
when folded for example -> + 15 
when unfolded -> Show less 
color default 30549F 
yes, with authors and keywords the same mechanics
3 lines of keywords and 3 lines of authors

**Comment by wujuu** [comment]
@Marcelinna @aleksandrahojszyk  Ok, I'll add the show more func for authors as well - for keywords (they are secondary tags) it's not implemented yet

**Comment by wujuu** [comment]
Added "show more" feat to authors. Changed from "+ X more" to "+ X". Changed the font weight on the button, because the color was the same as specified by @aleksandrahojszyk, although to me it still looks off

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**